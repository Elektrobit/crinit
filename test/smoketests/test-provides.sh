# shellcheck shell=bash
# SPDX-License-Identifier: MIT
#
# smoketest for crinit feature-based dependencies
#

setup() {
    crinit_config_setup
    rm -f /tmp/crinit-feature_dependent.log
}

run() {
    crinit_daemon_start "${SMOKETESTS_CONFDIR}"/demo.series
    sleep 3

    if ! "${BINDIR}"/crinit-ctl enable feature_provider; then
        echo "crinit-ctl enable of feature provider task failed"
        return 1
    fi
 
    sleep 1

    out="/tmp/crinit-feature_dependent.log"
    sample="${CMDPATH}"/test-"${SMOKETESTS_NAME}".out
    compare_output "$sample" "$out"
}

teardown() {
    # Terminate crinit daemon
    crinit_daemon_stop
}

