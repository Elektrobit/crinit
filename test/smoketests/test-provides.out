All feature-based dependencies have been fulfilled.
